core/utils/pqueue_tag.c

```diff
@@ -75,11 +64,21 @@ static void pqueue_tag_print_element(void* element) {
 //////////////////
 // Functions defined in pqueue_tag.h.
 
+int pqueue_tag_compare(pqueue_pri_t priority1, pqueue_pri_t priority2) {
+  return (lf_tag_compare(((pqueue_tag_element_t*)priority1)->tag, ((pqueue_tag_element_t*)priority2)->tag));
```

Comment:
Here an `unsigned long long` is casted to an `event_t *` this will result in warnings on any architecture where `sizeof(void *) != sizeof(unsigned long long)`

Reply:
I agree that pqueue_base should be changed so pqueue_pri_t is a void*.  This, however, requires redesigning the reaction queue, which has a (probably premature) optimization that combines the deadline and the level into a single index that it uses for ordering reactions.  I think this should be done as a separate PR.  Do you want to open an issue?  Probably we should benchmark the performance impact.  There is likely to be a cost.

Reply:
I think it is covered by this issue: https://github.com/lf-lang/reactor-c/issues/388